### PR TITLE
adapter: simplify compareByIdThenRev()

### PR DESCRIPTION
**Pull request by alxndrsn**
Remove one layer of functions.

### CODE REVIEW

File: packages/node_modules/pouchdb-core/src/adapter.js [hunk 1]
```diff
@@ -78,13 +74,12 @@ function cleanDocs(docs) {
 
 // compare two docs, first by _id then by _rev
 function compareByIdThenRev(a, b) {
-  var idCompare = compare(a._id, b._id);
-  if (idCompare !== 0) {
-    return idCompare;
+  if (a._id === b._id) {
+   const aStart = a._revisions ? a._revisions.start : 0;
+   const bStart = b._revisions ? b._revisions.start : 0;
+   return aStart - bStart;
```
Comment (garethbowen) [inline packages/node_modules/pouchdb-core/src/adapter.js:80]: The indentation is off I think...

```suggestion
    const aStart = a._revisions ? a._revisions.start : 0;
    const bStart = b._revisions ? b._revisions.start : 0;
    return aStart - bStart;
```

**Review by garethbowen — COMMENTED** [review]
Is this function used a lot? Does this yield a measurable improvement?

**Comment by alxndrsn** [comment]
> Is this function used a lot? Does this yield a measurable improvement?

I suspect not - it's specific to `bulkDocs(..., { ..., new_edits: true, ...})`

**Comment by SourceR85** [comment]
> > Is this function used a lot? Does this yield a measurable improvement?
> 
> I suspect not - it's specific to `bulkDocs(..., { ..., new_edits: true, ...})`

bulkDocs get called by every api-call to post and put... true is the default value for new_edits.
I think the relative performance impact depends on the use case.

**Review by garethbowen — APPROVED** [review]
I'm a little on the fence here... I think it's a little less readable, but should be a little faster. But I'm not one to stand in the way of progress, so approved!

**Pull request merged**